exporter/exporterbatcher/config.go

```diff
@@ -19,10 +19,29 @@ type Config struct {
 	// FlushTimeout sets the time after which a batch will be sent regardless of its size.
 	FlushTimeout time.Duration `mapstructure:"flush_timeout"`
 
+	SizeConfig `mapstructure:",squash"`
+
+	// Deprecated. Ignored if SizeConfig is set.
 	MinSizeConfig `mapstructure:",squash"`
+	// Deprecated. Ignored if SizeConfig is set.
 	MaxSizeConfig `mapstructure:",squash"`
 }
 
+type SizeConfig struct {
+	SizerType SizerType `mapstructure:",squash"`
+
+	MinSize int `mapstructure:"mix_size"`
+	MaxSize int `mapstructure:"max_size"`
+}
+
+type SizerType struct {
+	// Sizer should either be bytes or items.
+	Sizer string `mapstructure:"sizer"`
```

Comment:
make this private and create an `UnmarshalText` (implement this interface https://pkg.go.dev/encoding#TextUnmarshaler). See https://github.com/open-telemetry/opentelemetry-collector/blob/main/pipeline/internal/globalsignal/signal.go

That way, we can control to only allow specific values.